sdk/appcenter-data/src/test/java/com/microsoft/appcenter/data/DataTest.java

```diff
@@ -220,6 +224,13 @@ public ServiceCall answer(InvocationOnMock invocation) {
         assertEquals(1, docs.getCurrentPage().getItems().size());
         assertEquals(docs.getCurrentPage().getItems().get(0).getDeserializedValue().test, documents.get(0).getDeserializedValue().test);
 
+        /* Verify result was cached */
+        /*verify(mLocalDocumentStorage).writeOnline(
+                eq(USER_TABLE_NAME),
+                eq(documents.get(0)),
+                any(WriteOptions.class)
```

Comment:
We tried adding this verification but for some reason any(WriteOptions.class) fails and we tried several different solutions and nothing worked

Reply:
Will need to find a solution before merging, I can help.

Reply:
The problem is not WriteOptions.
Here the problem is using `eq(documents.get(0))` with an object where `equals` is not implemented to be deep equals.
Here it's enough to know the document id that was passed is actually the object under test: https://github.com/microsoft/appcenter-sdk-android/pull/1188/commits/63903501e49dbf563732197f8d7566bdce46561e